### PR TITLE
Fixes #10305: Add hover and onlick to dashboard caption

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/10305

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
PR rebased

File: rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js [hunk 1]
```diff
@@ -1793,123 +1793,3 @@ function createTable(gridId,data,columns, customParams, contextPath, refresh, st
   return table;
 }
 
-
-
-/**
- * The set of option that allows to configure chart js to be used as a sparkline.
- */
-function chartjsSparklineOption() {
-  return {
```
Comment (fanf) [inline rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js]: Why removing all that method? No more spark chart?

Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js]: Relocate from c3.js

File: rudder-web/src/main/webapp/javascript/rudder/homePage.js [hunk 2]
```diff
@@ -166,9 +168,86 @@ function doughnutChart (id,data,count, colors) {
...
+  // Highlight element
+  requestedElem.custom = requestedElem.custom || {};
+  requestedElem.custom.backgroundColor = Chart.helpers.getHoverColor(requestedElem._view.backgroundColor)
+  
+  // Add element
+  activeElements.push(requestedElem);
+  chart.tooltip._active = activeElements;
+  
+  chart.tooltip.update(true);
+  chart.update();
```
Comment (fanf) [inline rudder-web/src/main/webapp/javascript/rudder/homePage.js]: Why update() and then update(0)?

File: rudder-web/src/main/webapp/javascript/rudder/homePage.js [hunk 3]
```diff
@@ -166,9 +168,86 @@ function doughnutChart (id,data,count, colors) {
...
+//When mouse out of legend remove tooltip and highlight
+function closeTooltip(e ,index, name){
+  // Get chart
+  var chart = e.view[name];
+  
+  // Get active elements
+  var activeElements = chart.tooltip._active;
+  if(activeElements == undefined || activeElements.length == 0)
+    // No elements, nothing to do
+    return;
```
Comment (fanf) [inline rudder-web/src/main/webapp/javascript/rudder/homePage.js]: I didn't know js allowed if without {} and statement not on the immediate next line

File: rudder-web/src/main/webapp/javascript/rudder/homePage.js (hunk 2 above)
Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/javascript/rudder/homePage.js]: Ho..

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by fanf** [comment]
There is still a little problem on some case with the legend, but sufficiently inocuis to still get the benefits here.

**Comment by fanf** [comment]
Issue is: https://www.rudder-project.org/redmine/issues/10378

**Pull request merged**